orderer/consensus/etcdraft/chain.go

```diff
@@ -1235,6 +1237,39 @@ func pemToDER(pemBytes []byte, id uint64, certType string, logger *flogging.Fabr
 	return bl.Bytes, nil
 }
 
+// isConsensusUpgrade returns true if config block has a consensustype upgrade to smartbft.
+// in all other cases it returns false
+func (c *Chain) isConsensusUpgrade(block *common.Block) bool {
+	env := &common.Envelope{}
+	if err := proto.Unmarshal(block.Data.Data[0], env); err != nil {
+		return false
+	}
+
+	bundle, err := channelconfig.NewBundleFromEnvelope(env, c.CryptoProvider)
+	if err != nil {
+		return false
+	}
+
+	nextOrdererConfig, ok := bundle.OrdererConfig()
+	if ok {
+		if nextOrdererConfig.Capabilities().ConsensusTypeMigration() {
+			cfm := c.newConfigMetadata(block)
+			if cfm == nil {
+				return false
+			}
+
+			_, err := ComputeMembershipChanges(c.opts.BlockMetadata, c.opts.Consenters, cfm.Consenters)
```

Comment:
why do you compute membership changes? and then if the error is not nil, we continue? How do you know what is the reason it errors? 

Why not just look at the consensus type and at the blank metadata? why do you need `ComputeMembershipChanges` ?

Reply:
I wanted to use `ComputeMemberShip` to detect empty ConfigMetaData. This approach is buggy, So I have changed it now. I will do byte comparison of marshalled `ConfigMetaData` bytes with marshalled empty `ConfigMetaData` structure